libpromises/policy.c

```diff
@@ -451,7 +451,7 @@ Policy *PolicyMerge(Policy *a, Policy *b)
 
     if (extra_hashes != NULL)
     {
-        MapIterator it = MapIteratorInit((Map*) extra_hashes);
+        MapIterator it = MapIteratorInit(extra_hashes->impl);
```

Comment:
Using the impl member outside like this looks weird, I would expect `StringMapIteratorInit()` or something like that, but okay.

Reply:
Yeah, that's what I expected too. But that's out of scope of this change.